client/core/types.go

```diff
@@ -645,6 +660,10 @@ func (a *dexAccount) setupCryptoV2(creds *db.PrimaryCredentials, crypter encrypt
 	for i := 0; i < 8; i++ {
 		kids = append(kids, binary.LittleEndian.Uint32(byteSeq[i*4:i*4+4]))
 	}
+	// Last child is the account key index, which is 0 for the primary/default
+	// account and 0 > k < 2^31 for supplementary accounts where previously
+	// derived account keys have been cleared/reset.
+	kids = append(kids, keyIndex)
```

Comment:
Actually `0' + keyIndex` in the end or `2^31 < k < (2^32)-1` because `genChild` hardens it.

Reply:
I actually kinda think this comment is getting to far into the weeds.  I'd be good with just `// Last child is based on the account key index.` for this `append`.  The `genChild` docs can explain that hardened children are generated.